# ros_control_boilerplate Settings -----------------------
# Settings for ros_control control loop
generic_hw_control_loop:
  loop_hz: 25
  cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface
hardware_interface:
   joints:
      - test_servo
      - fr_drive
      - fl_drive
      - bl_drive
      - br_drive
      - intake
   sim_control_mode: 0 # 0: position, 1: velocity
   write_proto: 1 # 0 for bluetooth serial, 1 for websockets (nou2 vs 3)  
   port: /dev/rfcomm0
   websocket_uri: ws://192.168.4.1:9000
   websocket_logging_level: false # true is on, false off 

# Publish all joint states ----------------------------------
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: 50


fr_drive:
   type: motor
   port: 3
   inverted: false
   velocity_mult: 0.47
   velocity_feed_forward: 0.53

fl_drive:
   type: motor
   port: 2
   inverted: true
   velocity_mult: 0.47
   velocity_feed_forward: 0.53

br_drive:
   type: motor
   port: 4
   inverted: false
   velocity_mult: 0.47
   velocity_feed_forward: 0.53
   
bl_drive:
   type: motor
   port: 1
   inverted: true
   velocity_mult: 0.47
   velocity_feed_forward: 0.53

intake:
   type: motor
   port: 5
   inverted: true
   velocity_mult: 1.0
   velocity_feed_forward: 0.0

auto_motor:
   type: motor
   port: 32
   inverted: false
   velocity_mult: 0.0001
   velocity_feed_forward: 0.0

base_arm:
   type: servo
   port: 1
   inverted: false
   offset: 1.57 # 90 deg
   scale: 1.0
   initial_position: -0.94

second_stage_arm:
   type: servo
   port: 4
   inverted: false
   offset: 2.62 # 65 deg
   scale: 1.667
   initial_position: -2.19

test_servo:
   type: servo
   port: 1
   inverted: false
   offset: 0.0
   scale: 1.0
   initial_position: 0.0

# Individual Position Controllers ---------------------------------------
# Allows to send individual ROS msg of Float64 to each joint separately

# joint trajectory controller for arm
test_servo_controller:
  type: position_controllers/JointPositionController
  joint: test_servo

intake_controller:
  type: velocity_controllers/JointVelocityController
  joint: intake

auto_controller:
  type: velocity_controllers/JointVelocityController
  joint: auto_motor

mecanum_drive_controller:
  type: minimotor/MecanumDriveController
  # add parameters sometime maybe

initial:
   - group: armplangroup
     pose: rest

   